### PR TITLE
add last_commit property to Pagure project

### PR DESCRIPTION
**Pull request by sakalosj**

### CODE REVIEW

File: ogr/services/pagure/pull_request.py [hunk 1]
```diff
@@ -104,6 +104,10 @@ def created(self) -> datetime.datetime:
     def diff_url(self) -> str:
         return f"{self.url}#request_diff"
 
+    @property
+    def last_commit(self):
```
Comment (lachmanfrantisek) [inline ogr/services/pagure/pull_request.py]: Maybe a `head_commit` would fit better. (Last is a bit misleading because I don't know the direction...)

Comment (sakalosj) [reply ogr/services/pagure/pull_request.py]: changed

**Comment by sakalosj** [comment]
> Please:
> 
> * [x]  Add this to the abstract service class with `NotImplementedError`.
> * [x]  Add a test(s) for this.
> * [x]  And ideally, implement this for all three implementations. (I hope it will be very easy.) Otherwise, create an issue for that.

yes should be easy, but still takes time, so i opened issue for that, possibly will look at it later

**Comment by sakalosj** [comment]
> Thanks for the update.
> 
> To be honest, I would prefer using the same approach for tests as we do for everything else.
> 
> Just see the other existing tests how it's simple. Now, you actually reinventing the requre..;) I can help if you have some problems with that.

Its not reinventing requre, this is unit test it isolates tested method from all other stuff. In this case  method is using whole api reply stored in Pagure._raw_pr. It is enough to have reduced dictionary in test, but using full example you can just update `data_pagure_raw` fixture - replace value with parameter and use it other tests.

Requre is not unit testing tool by design.

**Comment by lachmanfrantisek** [comment]
@sakalosj requre will do exactly this for you -- save the Pagure response. It will be messy to have multiple ways how to write a test. (Mainly, it will be misleading for newcomers.)



You can probably only add one line to this test since the response is already saved:

https://github.com/packit-service/ogr/blob/fe64b280c0fd5ef15d79991c233c67c9888ae486/tests/integration/test_pagure.py#L287-L297


So: one new line vs. 209 new lines and new requirement to test the same thing

**Comment by sakalosj** [comment]
> @sakalosj requre will do exactly this for you -- save the Pagure response. It will be messy to have multiple ways how to write a test. (Mainly, it will be misleading for newcomers.)
> 
> You can probably only add one line to this test since the response is already saved:
> 
> https://github.com/packit-service/ogr/blob/fe64b280c0fd5ef15d79991c233c67c9888ae486/tests/integration/test_pagure.py#L287-L297
> 
> So: one new line vs. 209 new lines and new requirement to test the same thing

as mentioned before, Requre is not unit testing tool.

**Comment by sakalosj** [comment]
added also Requere test

File: tests/integration/test_pagure.py [hunk 2]
```diff
@@ -295,6 +295,7 @@ def test_pr_info(self):
             pr_info.diff_url
             == "https://pagure.io/ogr-tests/pull-request/5#request_diff"
         )
+        assert isinstance(pr_info.head_commit, str)
```
Comment (lachmanfrantisek) [inline tests/integration/test_pagure.py]: It does not differentiate the first/last commit.  (Or any other commit.) I would add the specific value as we do for all tests.

**Comment by sakalosj** [comment]
> Thanks!

welcome, and will also remove unnecessary imports

**Pull request merged**